### ISSUE TITLE
Extended Override syntax functions to draw random values for sweeps [Feature Request]

### ISSUE THREAD

**Issue by Halyjo** [facebookresearch/hydra#2415]
# 🚀 Feature Request

## Motivation

**Is your feature request related to a problem? Please describe.**

Assume we are training a deep learning model and want to tune the learning rate and the momentum for training. Further, assume we have an additional loss term that we also would like to tune, but which we will find out does not affect the performance at all. 

We run the following command:
`python train_my_model.py -m --learning_rate=0.001,0.01,0.1 --loss-term=0.1,0.5,0.9

This would result in training  3x3 = 9 models, one for every combination of learning rates and loss-terms. However, when we find out that the loss-term did not matter, we realize that all variations using learning rate 0.001 are equivalent, and the same for the others. We have therefore wasted 6 runs since the loss term did not matter. 

It would be very useful if we could write something like:

`python train_my_model.py -m --learning_rate=exponential(mu=0.01, 9) --loss-term=uniform(min=0, max=1, 9)`

The desired result here would be that we train the model 9 times again, but now each time picking a random value from an exponential distribution with mean 0.01 for the learning rate and a random value between 0 and 1 for the loss term. Looking at the results, we realize again that the loss term does not matter, but since all values were randomly chosen, we still have 9 different values for the learning rate and have wasted no runs. 

## Pitch

**Describe the solution you'd like**

When doing sweeps, having Extended Override syntax functions that includes a few random distribution functions would improve the usefulness of the command-line sweep support of hydra to avoid wasting runs when some parameters do not matter (see other functions https://hydra.cc/docs/advanced/override_grammar/extended/.). Most useful to me would be the functions: 
- `--uniform(<min>, <max>, <n>)`, where n is the number of sweeps, and min and max are the ends of the interval to draw from.  
- `--exponential(<mu>, <n>)`, where n is the number of sweeps and mu is the mean-parameter of the exponential distribution. 

**Describe alternatives you've considered**

One could always make a script that does this in python and just not use hydra here, but since the functionality is so close already with the range() function etc. using the extended override syntax (https://hydra.cc/docs/advanced/override_grammar/extended/) it would be very elegant to do it this way.

**Are you willing to open a pull request?** (See [CONTRIBUTING](../../CONTRIBUTING.md))
Unfortunately, no. I do not think I understand the code enough to do this and I do not have the time to dig into it now.

## Additional context
https://hydra.cc/docs/advanced/override_grammar/extended/

**Comment by Jasha10** [comment]
Hello @Halyjo,

Hydra supports "sweeper plugins" that implement this functionality. See the hydra docs on sweeper plugins. All three of Hydra's officially-supported sweepers (the [Ax sweeper](https://hydra.cc/docs/1.2/plugins/ax_sweeper/), the [Nevergrad sweeper](https://hydra.cc/docs/1.2/plugins/nevergrad_sweeper/) and the [optuna sweeper](https://hydra.cc/docs/1.2/plugins/optuna_sweeper/)) support randomly sampling from a uniform distribution or from a distribution on the log scale.

**Comment by Halyjo** [comment]
Hello,

Okay, perfect! 
Thank you and sorry for not checking better before requesting it.
